android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerButtonViewManager.kt

```diff
@@ -326,11 +330,35 @@ class RNGestureHandlerButtonViewManager : ViewGroupManager<ButtonViewGroup>(), R
       return false
     }
 
+    override fun onKeyUp(keyCode: Int, event: KeyEvent?): Boolean {
+      receivedKeyEvent = true
+      return super.onKeyUp(keyCode, event)
+    }
+
     override fun performClick(): Boolean {
       // don't preform click when a child button is pressed (mainly to prevent sound effect of
       // a parent button from playing)
-      return if (!isChildTouched() && soundResponder == this) {
-        soundResponder = null
+      return if (!isChildTouched()) {
+        tryFreeingResponder()
+
+        val gestureHandlerRootView = getGestureHandlerRootView()
+
+        when {
+          // Checks whether TalkBack is on
+          context.isScreenReaderOn() -> {
+            gestureHandlerRootView?.activateNativeHandlers(this)
+          }
+          receivedKeyEvent -> {
+            if (soundResponder == this) {
+              receivedKeyEvent = false
+            }
+            gestureHandlerRootView?.activateNativeHandlers(this)
+          }
+          soundResponder == this -> {
+            soundResponder = null
+          }
+        }
```

Comment:
```suggestion
        if (soundResponder == this) {
          soundResponder = null
        } else if (receivedKeyEvent || context.isScreenReaderOn()) {
          if (soundResponder == this) {
            receivedKeyEvent = false
          }
          
          getGestureHandlerRootView()?.activateNativeHandlers(this)
        }
```

I'm not sure if this is more readable in any way, but it should avoid traversing the view hierarchy in the most common branch.

Reply:
I wouldn't say that it's more readable, but avoiding unnecessary traversing sounds like a good reason to make that change.